plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterGeoJSON.java

```diff
@@ -0,0 +1,139 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jkiss.dbeaver.tools.transfer.stream.exporter;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.model.data.DBDAttributeBinding;
+import org.jkiss.dbeaver.model.exec.DBCSession;
+import org.jkiss.dbeaver.model.exec.DBCResultSet;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.tools.transfer.stream.IStreamDataExporter;
+import org.jkiss.dbeaver.tools.transfer.stream.IStreamDataExporterSite;
+import org.jkiss.dbeaver.model.data.json.JSONUtils;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+public class DataExporterGeoJSON implements IStreamDataExporter {
+
+    private OutputStreamWriter writer;
+    private PrintWriter out;
+    private List<DBDAttributeBinding> columns;
+    private boolean firstFeature = true;
+    private int geometryIndex = -1;
+    private IStreamDataExporterSite site;
+
+    @Override
+    public void init(IStreamDataExporterSite site) {
+        this.site = site;
+        this.writer = new OutputStreamWriter(site.getOutputStream(), StandardCharsets.UTF_8);
+        this.out = new PrintWriter(this.writer);
+        this.firstFeature = true;
+        this.geometryIndex = -1;
+        this.columns = null;
+    }
+
+    @Override
+    public void exportHeader(DBCSession session) throws IOException {
+        DBDAttributeBinding[] attrs = site.getAttributes();
+        this.columns = List.of(attrs);
+        // Find geometry column index by type (case-insensitive, contains "geometry", "geojson", "geography", or "geom")
+        for (int i = 0; i < columns.size(); i++) {
+            String typeName = columns.get(i).getTypeName();
+            if (typeName != null) {
+                String typeLower = typeName.toLowerCase();
+                if (typeLower.contains("geometry") || typeLower.contains("geojson") || typeLower.contains("geography") || typeLower.contains("geom")) {
+                    geometryIndex = i;
+                    break;
+                }
+            }
+        }
+        // Fallback: detect geometry column by common column names if type detection failed
+        if (geometryIndex < 0) {
+            for (int i = 0; i < columns.size(); i++) {
+                String colName = columns.get(i).getName();
+                if (colName != null) {
+                    String n = colName.trim().toLowerCase();
+                    if (n.equals("geometry") || n.equals("geom") || n.equals("the_geom") || n.equals("geography") || n.equals("geojson") || n.equals("shape") || n.equals("wkb_geometry")) {
+                        geometryIndex = i;
+                        break;
+                    }
+                }
+            }
+        }
+        out.write("{\"type\": \"FeatureCollection\", \"features\": [\n");
```

Comment:
GeoJson is a json. So maybe it is better to use JsonWriter that playing with strings?

As an example you can check TaskManagerImpl class where we use JsonWriter. It works in stream mode.